src/graph/validation/validate_neighbors.rs

```diff
@@ -0,0 +1,404 @@
+// License: MIT
+// Copyright © 2024 Frequenz Energy-as-a-Service GmbH
+
+//! Methods for validating that all components in a [`ComponentGraph`] are
+//! connected correctly.
+
+use crate::{
+    component_category::CategoryPredicates, ComponentCategory, Edge, Error, InverterType, Node,
+};
+
+use super::ComponentGraphValidator;
+
+impl<N, E> ComponentGraphValidator<'_, N, E>
+where
+    N: Node,
+    E: Edge,
+{
+    pub(super) fn validate_root(&self) -> Result<(), Error> {
+        self.ensure_root(self.root)?;
+        self.ensure_not_leaf(self.root)?;
+        self.ensure_exclusive_successors(self.root)?;
+
+        Ok(())
+    }
+
+    pub(super) fn validate_meters(&self) -> Result<(), Error> {
```

Comment:
Maybe document what kind validations are done here like with `is_xxx_meter()`? (same for other `validate_xxx()`.

Reply:
done